src/Adapter/Configuration.php

```diff
@@ -342,4 +342,12 @@ private function isStrict(?ShopConstraint $shopConstraint): bool
     {
         return null !== $shopConstraint ? $shopConstraint->isStrict() : false;
     }
+
+    /**
+     * @param string $key
+     */
+    public function deleteFromContext(string $key): void
```

Comment:
The only thing that bothers me here is that it's a "context function" that depends on the value set in the Context, although it does the job it would be safer not to reproduce the same mistake in this Configuration service

So we should use a `ShopConstraint` parameter to target the specific value that we aim to delete, and since in the `updateConfigurationValue` which uses this method you already have it, it shouldn't be a big modif

Reply:
And this would allow to remove all the overrides from Shop and ShopGroup context (with several consecutive calls) regardless of which state the Context is in

Reply:
So if I follow you @jolelievre we need to do
```
    public static function deleteFromContext($key)
```
⬇️ 
```
    public static function deleteFromContext($key, $idShopGroup = null, $idShop = null)
```